pkg/issuer/acme/dns/dnsmadeeasy/dnsmadeeasy.go

```diff
@@ -0,0 +1,265 @@
+func (d *DNSProvider) Present(domainName, token, keyAuth string) error {
+	glog.Errorf("d.dns01Nameservers: %v", d.dns01Nameservers)
+	fqdn, value, ttl, err := util.DNS01Record(domainName, keyAuth, d.dns01Nameservers)
+	if err != nil {
+		return err
+	}
+
+	authZone, err := util.FindZoneByFqdn(fqdn, util.RecursiveNameservers)
+	if err != nil {
+		return err
+	}
+
+	// fetch the domain details
+	domain, err := d.getDomain(authZone)
+	if err != nil {
+		return err
+	}
+
+	// validate against domainID
+	if d.domainID > 0 && d.domainID != uint(domain.ID) {
+		return fmt.Errorf("DNS Made Easy domain ID %d is not authoritative for %s", d.domainID, authZone)
+	}
+
+	// find existing records, as there is no update-or-create
+	name := strings.Replace(fqdn, "."+authZone, "", 1)
+	records, err := d.getRecords(domain, name, "TXT")
+	if err != nil {
+		return err
+	}
+
+	// check for existing correct record
+	for _, record := range *records {
+		if record.Value == value {
+			return nil
+		}
+	}
+
+	// create new record
+	record := &Record{Type: "TXT", Name: name, Value: value, TTL: uint(ttl)}
+	err = d.createRecord(domain, record)
+	return err
+}
+
+// CleanUp removes the TXT records matching the specified parameters
+func (d *DNSProvider) CleanUp(domainName, token, keyAuth string) error {
+	fqdn, _, _, err := util.DNS01Record(domainName, keyAuth, d.dns01Nameservers)
+	if err != nil {
+		return err
+	}
+
+	authZone, err := util.FindZoneByFqdn(fqdn, util.RecursiveNameservers)
+	if err != nil {
+		return err
+	}
+
+	// fetch the domain details
+	domain, err := d.getDomain(authZone)
+	if err != nil {
+		return err
+	}
+
+	// validate against domainID
+	if d.domainID > 0 && d.domainID != domain.ID {
+		return fmt.Errorf("DNS Made Easy domain ID %n is not authoritative for %s", d.domainID, authZone)
+	}
+
+	// find matching records
+	name := strings.Replace(fqdn, "."+authZone, "", 1)
+	records, err := d.getRecords(domain, name, "TXT")
+	if err != nil {
+		return err
+	}
+
+	// delete records
+	for _, record := range *records {
+		err = d.deleteRecord(record)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (d *DNSProvider) Timeout() (timeout, interval time.Duration) {
+	return 120 * time.Second, 2 * time.Second
+}
+
+func (d *DNSProvider) getDomain(authZone string) (*Domain, error) {
+	domainName := authZone[0 : len(authZone)-1]
+	resource := fmt.Sprintf("%s%s", "/dns/managed/name?domainname=", domainName)
+
+	resp, err := d.sendRequest(http.MethodGet, resource, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	domain := &Domain{}
+	err = json.NewDecoder(resp.Body).Decode(&domain)
+	if err != nil {
+		return nil, err
+	}
+
+	return domain, nil
+}
+
+func (d *DNSProvider) getRecords(domain *Domain, recordName, recordType string) (*[]Record, error) {
+	resource := fmt.Sprintf("%s/%d/%s%s%s%s", "/dns/managed", domain.ID, "records?recordName=", recordName, "&type=", recordType)
```

Comment:
Adjust this string to avoid the %s%s%s%s at the end?

i.e.:

```
fmt.Sprintf("/dns/managed/%d/records?recordName=%s&type=%s", domain.ID, recordName, recordType)
```

Reply:
Better yet - you can switch to build URLs/query paths using `url.URL` (which should also handle URL encoding automatically for you)